### PR TITLE
Add data prop to FService::Base#Check

### LINKED ISSUE #19: Add support to custom data on Base#Check
Hi,

So, myself and my colleagues are always complaining that we cannot pass a custom data to the `Base#Check` and always need to write both the Success and Failure to be able to use a custom data. I think it is a good idea to keep the actual behavior to validation purposes, but also expand it's usage by having an optional data argument.

Or even better, having a different data prop depending on the actual return could be a good one too, what do you think?

PR: #18

### PR DESCRIPTION
**Pull request by andre-filho**
hey,

everytime myself and my colleages use the `FService::Base#Check` method we end up with the problem of not being able to pass a custom data to the resulting `Result`. In the end, we end up writing code very similar to what the gem does in the method definition.

Closes #19

### CODE REVIEW

File: lib/f_service/base.rb [hunk 1]
```diff
@@ -157,14 +158,21 @@ def Failure(type = nil, data: nil)
...
     # @return [Result::Success, Result::Failure] a Result from the boolean expression
-    def Check(type = nil)
+    def Check(type = nil, data: nil)
       res = yield
 
+      unless data.nil?
+        return res ? Success(type, data: data) : Failure(type, data: data)
+      end
+
       res ? Success(type, data: res) : Failure(type, data: res)
```
Comment (MatheusRich) [inline lib/f_service/base.rb:176]:
```suggestion
      final_data = data || res
      res ? Success(type, data: final_data) : Failure(type, data: final_data)
```

**Comment by MatheusRich** [comment]
Nice addition! I was planning to add this very patch! Thanks!

File: lib/f_service/base.rb [hunk 2]
```diff
@@ -146,8 +146,9 @@ def Failure(type = nil, data: nil)
 
     # Converts a boolean to a Result.
     # Truthy values map to Success, and falsey values map to Failures.
-    # You can optionally provide a type for the result. The result value
-    # is the evaluated value of the given block.
+    # You can optionally provide a type for the result.
+    # The result value defaults as the evaluated value of the given block,
+    # if you want another value you can pass it through the data argument.
```
Comment (MatheusRich) [inline lib/f_service/base.rb:151]:
```suggestion
    # The result value defaults as the evaluated value of the given block.
    # If you want another value you can pass it through the `data:` argument.
```

File: lib/f_service/base.rb [hunk 3]
```diff
@@ -157,14 +158,21 @@ def Failure(type = nil, data: nil)
     #
     #       Check(:math_works) { 1 > 2 }
     #       # => #<Failure @error=false, @type=:math_works>
+    #
+    #       Check(:math_works, data: 1 + 2) { 1 > 2 }
```
Comment (MatheusRich) [inline lib/f_service/base.rb:162]: We should add an example for the Success too

Comment (andre-filho) [reply lib/f_service/base.rb:162]: Working on it

Comment (andre-filho) [reply lib/f_service/base.rb:162]: done

**Pull request merged**